lib/psych/comments/node_ext.rb

```diff
@@ -5,7 +5,16 @@ def leading_comments
     @leading_comments ||= []
   end
 
+  def line_end_comments
+    @line_end_comments ||= []
+  end
+
   def trailing_comments
     @trailing_comments ||= []
   end
+
+  alias end_column_without_comment end_column
+  def end_column
+    end_column_without_comment + (line_end_comments[0]&.length || 0)
+  end
```

Comment:
This change has two problems:

- The effect of this overriding is somewhat unpredictable when nodes with comments are passed to a tool unaware of comments.
- The behavior is inconsistent with the other parts of this library, as only line_end_comments are honored.

Could you drop this change for now? I'm willing to discuss this change after the PR is merged. Though in my opinion, adding newly named methods seems better than overriding existing ones.